mewarpx/mewarpx/plotting.py

```diff
@@ -0,0 +1,661 @@
+
+    def _gen_plot_contours(self):
+        """Generate the list of contours."""
+        if self.params['scale'] == 'linear':
+            return np.linspace(self.valmin, self.valmax,
+                               self.params['ncontours'])
+        # Make sure end points are captured, rather than missed due to floating
+        # point errors.
+        pmin = self.valmin - 1e-5*(self.valmax - self.valmin)
+        pthresh = np.abs(self.params["linthresh"])
+        pmax = self.valmax + 1e-5*(self.valmax - self.valmin)
+        # Try to have a proportional set of negative log contours, linear
+        # contours, and positive log contours. Linear is always 'linportion' of
+        # total # of contours here.
+        # Plot linear contours (any values in linear range)?
+        if pmin < pthresh and pmax > -pthresh:
+            lincontours = np.linspace(
+                max(pmin, -pthresh), min(pmax, pthresh),
+                int(round(self.params["ncontours"]*self.params["linportion"])))
+        else:
+            lincontours = np.array([])
+        nlogcontours = int(round(self.params["ncontours"] - len(lincontours)))
+        # Plot negative logarithmic contours?
+        if pmin < -pthresh:
+            nnegcontours = max(0, int(round(
+                ((np.log(-pmin) - np.log(pthresh))
+                 / (np.log(-pmin) + np.log(max(pthresh, abs(pmax)))
+                    - 2.*np.log(pthresh))
+                 * nlogcontours))))
+            neglogcontours = -1.0*np.exp(
+                np.linspace(np.log(pthresh), np.log(abs(pmin)), nnegcontours))
+        else:
+            nnegcontours = 0
+            neglogcontours = np.array([])
+        # Plot positive logarithmic contours?
+        if pmax > pthresh:
+            nposcontours = max(0, nlogcontours - nnegcontours)
+            poslogcontours = np.exp(np.linspace(np.log(pthresh), np.log(pmax),
+                                                nposcontours))
+        else:
+            poslogcontours = np.array([])
+        # Multiply by the appropriate factor, eliminate duplicates with set,
+        # then sort
+        contour_points = sorted(set(
+            np.concatenate([neglogcontours, lincontours, poslogcontours])))
+        return contour_points
+
+
+class SimInfo(object):
+
+    """Store basic simulation parameters used throughout analysis.
+    This must contain:
+        SimInfo.nxyz (int nx, int ny, int nz)
+        SimInfo.pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax)
+        SimInfo.geom (str 'XZ', 'RZ' or 'XYZ')
+        SimInfo.dt (float)
+        SimInfo.periodic (bool)
+    Note:
+        This base class has been created to provide additional plotting
+        functionality, and allow post-run creation of this object if needed.
+    """
+
+    def __init__(self, nxyz, pos_lims, geom, dt, periodic=True):
+        self.nxyz = nxyz
+        self.pos_lims = pos_lims
+        self.geom = geom
+        self.dt = dt
```

Comment:
We should be able to get all this information from importing the `mwxrun` object.